### PR TITLE
refactoring costmaps to use parameters

### PR DESCRIPTION
**Pull request by NFiedler**
## Proposed changes
refactoring costmaps to use parameters and class fields

## Necessary checks
- [ ] Update package version
- [ ] Run `catkin build`
- [ ] Write documentation
- [ ] Create issues for future work
- [ ] Test on your machine
- [ ] Test on the robot
- [ ] Put the PR on our Project board

### CODE REVIEW

File: bitbots_blackboard/src/bitbots_blackboard/capsules/world_model_capsule.py [hunk 1]
```diff
@@ -10,6 +10,7 @@
 from scipy.ndimage import gaussian_filter
 import matplotlib.pyplot as plt
 from scipy.interpolate import griddata
+import PIL
```
Comment (johagge) [inline bitbots_blackboard/src/bitbots_blackboard/capsules/world_model_capsule.py:13]: unused import?

File: bitbots_blackboard/src/bitbots_blackboard/capsules/world_model_capsule.py [hunk 2]
```diff
@@ -415,18 +422,21 @@ def global_costmap_update_callback(self, msg):
     def add_obstacles_costmap(self):
         scale = 1 / self.local_obstacle_map.info.resolution / 2
         obstacle = self.local_obstacle_map_numpy.T[
-                    int(self.local_obstacle_map_numpy.shape[0] / 2 - 9 * scale) : int(self.local_obstacle_map_numpy.shape[0] / 2 + 9 * scale) : 2,
-                    int(self.local_obstacle_map_numpy.shape[1] / 2 - 6 * scale) : int(self.local_obstacle_map_numpy.shape[1] / 2 + 6 * scale) : 2].astype(np.float) / 100 * 0.4
+                    int(self.local_obstacle_map_numpy.shape[0] / 2 - self.field_length * scale) : int(self.local_obstacle_map_numpy.shape[0] / 2 + self.field_length * scale) : 2,
+                    int(self.local_obstacle_map_numpy.shape[1] / 2 - self.field_width * scale) : int(self.local_obstacle_map_numpy.shape[1] / 2 + self.field_width * scale) : 2].astype(np.float) / 100 * 0.4
```
Comment (johagge) [inline bitbots_blackboard/src/bitbots_blackboard/capsules/world_model_capsule.py:426]:
```suggestion
                    int(self.local_obstacle_map_numpy.shape[1] / 2 - self.field_width * scale) : int(self.local_obstacle_map_numpy.shape[1] / 2 + self.field_width * scale) : 2].astype(np.float) / 100 * 0.4 # 0.4 is the maximum cost of one obstacle
```

File: bitbots_blackboard/src/bitbots_blackboard/capsules/world_model_capsule.py [hunk 3]
```diff
@@ -538,20 +540,14 @@ def get_gradient_direction_at_field_position(self, x, y):
         :param x: Field coordiante in the x direction
         :param y: Field coordiante in the y direction
         """
+        # for debugging only
         if self.costmap.sum() > 0 and False:
```
Comment (johagge) [inline bitbots_blackboard/src/bitbots_blackboard/capsules/world_model_capsule.py:544]:
```suggestion
        if False and self.costmap.sum() > 0:
```

**Review by johagge — APPROVED** [review]

File: bitbots_blackboard/src/bitbots_blackboard/capsules/world_model_capsule.py (hunk 1 above)
Comment (NFiedler) [reply bitbots_blackboard/src/bitbots_blackboard/capsules/world_model_capsule.py:13]: will be used in next PR

**Pull request merged**